### PR TITLE
streamingest: assert ingest processors cover whole tenant

### PR DESCRIPTION
**Pull request by dt**
Previously we only tracked, in the ingest frontier, the set of spans being streamed by each proc.

Now we track tenant span, and during planning we assert the former adds up to the latter.

### CODE REVIEW

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go [hunk 1]
```diff
@@ -808,7 +807,12 @@ func constructStreamIngestionPlanSpecs(
 			DestInstanceID:    destID,
 		}
 		streamIngestionSpecs[destID].PartitionSpecs[partition.ID] = partSpec
-		trackedSpans = append(trackedSpans, partition.Spans...)
+		spanGroup.Add(partition.Spans...)
+	}
+
+	tenantSpan := keys.MakeTenantSpan(sourceTenantID)
+	if !spanGroup.Encloses(tenantSpan) {
+		return nil, nil, errors.Newf("span %s not covered by %s!", tenantSpan, spanGroup.Slice())
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go:815]: AssertionFailedf?

**Review by stevendanna — APPROVED** [review]

**Comment by dt** [comment]
TFTR!

bors r+

**Comment by rickystewart** [comment]
Wow, lots of pre-emptions on that `bors` run.

bors r+

**Comment by dt** [comment]
blathers backport 24.1

**Pull request merged**